datasource.go

```diff
@@ -0,0 +1,204 @@
+package main
```

Comment:
It's up to you. For this PR, I thought it was simplest to just include the two go files that I needed. Technically they can be merged into a single file.

WRT vendor, it's up to what we should do for this project. Since the go.mod and go.sum is committed it will always download the correct vendors. Go mod stores the deps into the local `$GOPATH` cache and runs differently from the traditional vendors so they should not be conflated and I would be committing the cache if you want a copy in this repo.